### DISCUSSION TITLE
Your Great Software on ESP S2 Wrover Saloa

### DISCUSSION THREAD

**Discussion by Micha2007fe** [schreibfaul1/ESP32-audioI2S#297] [Q&A] [answered]
Hello,
iám Michaela an yopung girl which is working with ESP.
I found this great Software and used it on a ESP32 U2 Wroom until now.
Now I want to have the PSRAM Support and I  looking for 4MB SRAM for the 32U ESP.
Because I do not found I ordered a ESP32 S2 Wrover ( Saloa)

But on this the Arduino Compiler brings out error codes.
Iam so sad , and Iam trying su much.

Now, I registered here in the hope anybody can help me to get this great SOftware go
working on my ESP32 S2 Wrover Saloa.

Arduiono IDE is actually, Bib und Libs are up to date and I have Windows 10 64 Bit.
Compiling with ESP32 works
Compiling with ESP Wrover works
Only Compiling with ESP S2 bring out error codes:

`
In file included from C:\Users\Michaela\Desktop\Arduino_Micha\26_ESP32_Webradio\ESP32_Webradio\ESP32_Webradio.ino:14:
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:157:37: error: 'i2s_dac_mode_t' has not been declared
     Audio(bool internalDAC = false, i2s_dac_mode_t channelEnabled = I2S_DAC_CHANNEL_LEFT_EN); // #99
                                     ^~~~~~~~~~~~~~
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:194:64: error: 'i2s_dac_mode_t' has not been declared
     [[deprecated]]void setInternalDAC(bool internalDAC = true, i2s_dac_mode_t channelEnabled = I2S_DAC_CHANNEL_LEFT_EN);
                                                                ^~~~~~~~~~~~~~
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:471:5: error: 'i2s_dac_mode_t' does not name a type; did you mean 'i2s_mode_t'?
     i2s_dac_mode_t  m_f_channelEnabled = I2S_DAC_CHANNEL_LEFT_EN;  // internal DAC on GPIO26 for M5StickC/Plus
     ^~~~~~~~~~~~~~
     i2s_mode_t
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:157:69: error: 'I2S_DAC_CHANNEL_LEFT_EN' was not declared in this scope
     Audio(bool internalDAC = false, i2s_dac_mode_t channelEnabled = I2S_DAC_CHANNEL_LEFT_EN); // #99
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:157:69: note: suggested alternative: 'I2S_DATA_ENABLE_TEST_EN'
     Audio(bool internalDAC = false, i2s_dac_mode_t channelEnabled = I2S_DAC_CHANNEL_LEFT_EN); // #99
                                                                     ^~~~~~~~~~~~~~~~~~~~~~~
                                                                     I2S_DATA_ENABLE_TEST_EN
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:194:96: error: 'I2S_DAC_CHANNEL_LEFT_EN' was not declared in this scope
     [[deprecated]]void setInternalDAC(bool internalDAC = true, i2s_dac_mode_t channelEnabled = I2S_DAC_CHANNEL_LEFT_EN);
                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~
C:\Users\Michaela\Desktop\Arduino_Micha\libraries\ESP32-audioI2S-master\src/Audio.h:194:96: note: suggested alternative: 'I2S_DATA_ENABLE_TEST_EN'
     [[deprecated]]void setInternalDAC(bool internalDAC = true, i2s_dac_mode_t channelEnabled = I2S_DAC_CHANNEL_LEFT_EN);
                                                                                                ^~~~~~~~~~~~~~~~~~~~~~~
                                                                                                I2S_DATA_ENABLE_TEST_EN
`

**Comment by schreibfaul1** [comment]
Hello Michaela,
in the ESP32 core 0 is used for the background functions, on core 1 the Arduino 'loop' runs by default. Unfortunately, the S2 only has one core. Even without compiler errors, the audio output would not run smoothly on the S2.

**Comment by Micha2007fe** [comment, marked as answer]
Dear schreibfaul1
many thanks for this Info!
So I go to buy a wrover Board with dual Core and PSRAM.

I was looking for SPI SRAM for the ESP32 Wroom. but I do not found
in Germany any fitting @ Conrad, Ebay etc...
Digikey is to expensive for only 1 or two piece.
Dees anyboday has maybe a tip were to get a SO8 SPI SRAM in 3,3V ?

**Reply by schreibfaul1** [reply]
I don't use many different boards. I can recommend this:https://www.amazon.de/DollaTek-ESP32-Psram-Bluetooth-ESP32-Wrover/dp/B07RWHK65X/ The SD card adapter can be also used for SD_MMC.

Or that: https://www.amazon.de/Gernian-Seriell-Esp32-Aduio-Entwicklung-Esp32-A1S-schwarz/dp/B09RF4P9BQ/ref=sr_1_1 that has the DAC and amplifiers on the board
